spinn_front_end_common/interface/front_end_common_interface_functions.py

```diff
@@ -498,21 +504,22 @@ def start_all_cores(self, executable_targets, app_id, no_full_runs):
                     "Only {} of {} processors started:{}"
                     .format(processors_running, total_processors, break_down))
 
-    def wait_for_execution_to_complete(
-            self, executable_targets, app_id, runtime, time_scaling):
+    def wait_for_execution_to_complete(self, executable_targets, app_id,
+        runtime, time_scaling, no_full_runs):
         """
 
         :param executable_targets:
         :param app_id:
         :param runtime:
         :param time_scaling:
+        :param no_full_runs: the number of runs been done between setup and end
         :return:
         """
 
         total_processors = executable_targets.total_processors
         all_core_subsets = executable_targets.all_core_subsets
 
-        time_to_wait = ((runtime * time_scaling) / 1000.0) + 1.0
+        time_to_wait = ((runtime * time_scaling) / 1000.0) * 0.9
```

Comment:
Again, if you run the simulation for 0.1 seconds and wait for runtime plus one second, you are waiting for more time than the simulation is actually running. However, multiplying with 0.9 is not great for longer runtimes. Ideally, the host could enter some kind of waiting state and be notified by the board once the simulation is completed.